### PR TITLE
(openstack) provide support for swift-based user data

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy [hunk 1]
```diff
@@ -165,12 +166,16 @@ class DeployOpenstackAtomicOperation implements AtomicOperation<DeploymentResult
...
-      String userData
-      if (description.userData) {
-        //TODO this will change to resolve based on userDataType
-        if (description.userData.startsWith("http")) {
+      String userData = ""
+      if (description.userDataType && description.userData) {
+        if (UserDataType.fromString(description.userDataType) == UserDataType.URL) {
           task.updateStatus BASE_PHASE, "Resolving user data from url $description.userData..."
           userData = description.userData.toURL()?.text
+        } else if(UserDataType.fromString(description.userDataType) == UserDataType.SWIFT) {
```
Comment (anotherchrisberry) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy]: missing a space between `if` and `(` but, if nothing's complaining, I won't.

**Comment by anotherchrisberry** [comment]
I won't pretend to know what this is, but it won't break anything on our end so I'm happy to merge it unless you want someone else to review it.

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy (hunk 1 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/servergroup/DeployOpenstackAtomicOperation.groovy]: fixed

**Comment by drmaas** [comment]
Swift is an openstack component that allows you to store and download arbitrary binary data.

**Pull request merged**